### ISSUE TITLE
Performance issue with Get-NavContainerAppInfo

### ISSUE THREAD

**Issue by fridrichovsky** [microsoft/navcontainerhelper#2591]
**Describe the issue**
I have performance issue with function "Get-NavContainerAppInfo". Problem is when I try get sorted applications and when I have more versions for Microsoft applications. The function repeats the search for dependencies (it can take more than 1hour). I located problem in this part:

```
        function AddDependency { Param($dependency)
            #Write-Host "Add Dependency $($dependency.Name) $($dependency.Version)"
            $dependentApp = $apps | Where-Object { $_.AppId -eq $dependency.AppId  }
            if ($dependentApp) {
                AddAnApp -AnApp $dependentApp
            }
```

Problem is that "$dependentApp" content more applications. So function "AddAnApp" can't skip knowen appications. I changed the code to this:
```
        function AddDependency { Param($dependency)
            #Write-Host "Add Dependency $($dependency.Name) $($dependency.Version)"
            $dependentApp = $apps | Where-Object { $_.AppId -eq $dependency.AppId  }
            if ($dependentApp) {
                @($dependentApp) | ForEach-Object { AddAnApp -AnApp $_ }
            }
```
This helped me with performance issue and I can get sorted applications.

If you agree with this change, should I make a PR for this change?

**Comment by freddydk** [comment]
Is this because you have multiple versions of the same app published?
If you add the parameter -installedOnly - I assume that the problem is fixed as well?
I think your fix is fine, but I assume that you might want to use installedOnly

**Comment by fridrichovsky** [comment]
Yes problem is that I have multiple verisions of the same app. 
I can use intalled only. It works as you wrote. I am using script for installation application with dependent application. When I get error during installation I need rerun my script. So I am state when applications is not installed. I need get same list of applications. It is reason why I have to use function for all published extension.

**Comment by freddydk** [comment]
Shipped in the latest prerelease

**Comment by freddydk** [comment]
Shipped in 4.0.3
